### DISCUSSION TITLE
Status of nodejs-mobile

### DISCUSSION THREAD

**Discussion by tobias-klein** [ezra-bible-app/ezra-bible-app#498] [Development]
It seems like there are some changes for the maintenance of nodejs-mobile.

This is what they say in their [repo](https://github.com/JaneaSystems/nodejs-mobile):

![image](https://user-images.githubusercontent.com/17739895/139137094-8f9979f8-225c-46af-aecf-39a2be60283a.png)

However, it seems like the intention is to still allow maintenance under a more public model. They are now pointing to this repo:
https://github.com/nodejs-mobile/nodejs-mobile

If that project fails to maintain this software in the future we need to look for a new way to implement the backend on Android/Cordova.

This is just for information at this point and right now things are still working fine with nodejs-mobile (even though it's a bit rough to set up a functioning development environment). But there is a risk that at some point we may need something new.

**Comment by mochsner** [comment]
I noticed on the https://github.com/nodejs-mobile/nodejs-mobile README, there's iOS support as well. I don't use iOS, but being on both might help gain adoption (e.g. being standardized for Bible studies, etc.).

Anyway, my question - is there a technical barrier to entry for getting onto the Apple App Store you've ran into, or is it simply their tax on the process (now maybe $199, but used to be $99/year)? 

If there's a technical barrier to entry, it's definitely worth being aware of or documenting here - as there's other bundling packages (like Ionic) that might be advantageous to research which have pretty good support for it.

**Reply by tobias-klein** [reply]
Yes, nodejs-mobile does support iOS. I just never bothered to pursue this further. It's technically another platform and another release bundle to add during packaging for every release.

Ionic may be an option in the future.

However, the backend functionality must be somehow supported:
- Sword library with node-sword-interface
- sqlite
- ideally nodejs - otherwise we need to have two different backends for desktop vs mobile

The maintainer over at https://github.com/nodejs-mobile/nodejs-mobile has actually resumed maintenance. However, the Cordova plugin of nodejs-mobile is still unmaintained.

For the time being, my build of Ezra based on the current Cordova + nodejs-mobile technology is working. Whenever it stops working, I'll consider an upgrade (based on me looking into the Cordova plugin of nodejs-mobile) or an alternative approach.

Something that Troy, the maintainer of [Bishop](https://git.crosswire.org/main/bishop) is doing: He created his own Cordova plugin based on Sword. In case of Ezra, I would not only need that, but also sqlite.

Even though the build process is somehow difficult to reproduce right now ... it's quite convenient, because I can re-use the exact same node-sword-interface software and all nodejs-backend functionalities on Android and there are only minor changes compared to the desktop variant.

Regarding entry barriers for macOS / iOS builds ... I actually do pay the 99$ per year, to be able to notarize the Ezra builds for macOS. The same developer license would probably enable me to produce iOS builds without additional cost ...

**Comment by tobias-klein** [comment]
We have meanwhile migrated to a community-maintained version of nodejs-mobile. This discussion is therefore no longer relevant.
